lib/scholar/metrics.ex

```diff
@@ -331,6 +331,49 @@ defmodule Scholar.Metrics do
     Nx.divide(true_negatives, false_positives + true_negatives + 1.0e-16)
   end
 
+  @doc ~S"""
+  Calculates the confusion matrix.
+
+  ## Argument Shapes
+
+    * `y_true` - $\(d_0, d_1, ..., d_n\)$
+    * `y_pred` - $\(d_0, d_1, ..., d_n\)$
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([0, 0, 1, 1, 2, 2], type: {:u, 32})
+      iex> y_pred = Nx.tensor([0, 1, 0, 2, 2, 2], type: {:u, 32})
+      iex> Scholar.Metrics.confusion_matrix(y_true, y_pred, num_classes: 3)
+      #Nx.Tensor<
+        s64[3][3]
+        [
+          [1, 1, 0],
+          [1, 0, 1],
+          [0, 0, 2]
+        ]
+      >
+  """
+  def confusion_matrix(y_true, y_pred, opts \\ []) do
+    num_classes = opts[:num_classes]
+
+    cm =
+      Nx.iota({num_classes, num_classes})
+      |> then(&Nx.subtract(&1, &1))
```

Comment:
```suggestion
    cm = Nx.broadcast(0, {num_classes, num_classes})
```

Reply:
We should probably also pass the backend from either y_true or y_pred. Not exactly sure if defn will pick up on that